server/src/main/java/fr/uga/l3miage/tp4/repositories/CandidateRepository.java

```diff
@@ -0,0 +1,14 @@
+package fr.uga.l3miage.tp4.repositories;
+
+import fr.uga.l3miage.tp4.models.CandidateEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.util.Set;
+
+@Repository
+public interface CandidateRepository extends JpaRepository<CandidateEntity, Long> {
+    Set<CandidateEntity> findCandidateEntitiesByTestCenterContains(String code);
```

Comment:
ça ne marche pas, car dans votre implémentation, puisque vous avez pas déclarer le `@Enummarated` alors vous avez un entier en base et pas un string, donc vous pouvez pas faire un like dessus. Mais vous auriez pu comparer directement avec un TestCenter code. Sinon vous chercher tous les éléments donc `findAll`
```suggestion
    Set<CandidateEntity> findAllCandidateEntitiesByTestCenterCode(TestCenterCode code);
```